x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/action/UpdateConnectorLastSyncStatsAction.java

```diff
@@ -0,0 +1,247 @@
+
+import org.elasticsearch.ElasticsearchParseException;
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.action.ActionResponse;
+import org.elasticsearch.action.ActionType;
+import org.elasticsearch.action.DocWriteResponse;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.xcontent.XContentHelper;
+import org.elasticsearch.rest.RestStatus;
+import org.elasticsearch.xcontent.ConstructingObjectParser;
+import org.elasticsearch.xcontent.ObjectParser;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentParser;
+import org.elasticsearch.xcontent.XContentParserConfiguration;
+import org.elasticsearch.xcontent.XContentType;
+import org.elasticsearch.xpack.application.connector.ConnectorSyncInfo;
+import org.elasticsearch.xpack.application.connector.ConnectorSyncStatus;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.util.Objects;
+
+import static org.elasticsearch.action.ValidateActions.addValidationError;
+import static org.elasticsearch.xcontent.ConstructingObjectParser.optionalConstructorArg;
+
+public class UpdateConnectorLastSyncStatsAction extends ActionType<UpdateConnectorLastSyncStatsAction.Response> {
+
+    public static final UpdateConnectorLastSyncStatsAction INSTANCE = new UpdateConnectorLastSyncStatsAction();
+    public static final String NAME = "cluster:admin/xpack/connector/update_last_sync_stats";
+
+    public UpdateConnectorLastSyncStatsAction() {
+        super(NAME, UpdateConnectorLastSyncStatsAction.Response::new);
+    }
+
+    public static class Request extends ActionRequest implements ToXContentObject {
+
+        private final String connectorId;
+
+        private final ConnectorSyncInfo syncInfo;
+
+        public Request(String connectorId, ConnectorSyncInfo syncInfo) {
+            this.connectorId = connectorId;
+            this.syncInfo = syncInfo;
+        }
+
+        public Request(StreamInput in) throws IOException {
+            super(in);
+            this.connectorId = in.readString();
+            this.syncInfo = in.readOptionalWriteable(ConnectorSyncInfo::new);
+        }
+
+        public String getConnectorId() {
+            return connectorId;
+        }
+
+        public ConnectorSyncInfo getSyncInfo() {
+            return syncInfo;
+        }
+
+        @Override
+        public ActionRequestValidationException validate() {
+            ActionRequestValidationException validationException = null;
+
+            if (Strings.isNullOrEmpty(connectorId)) {
+                validationException = addValidationError("[connector_id] cannot be null or empty.", validationException);
```

Comment:
Same here, this _could_ live inside a shared class, but also not blocking. We could also create a refactoring issue for these things, if we want to.